airbyte-cdk/python/airbyte_cdk/utils/schema_inferrer.py

```diff
@@ -22,13 +23,25 @@ def to_schema(self):
         return schema
 
 
+class IntegerToNumber(Number):
+    """
+    This class has the regular Number behaviour, but it will never emit an integer type.
+    """
+
+    def __init__(self, node_class):
+        super(IntegerToNumber, self).__init__(node_class)
+        self._type = "number"
+
+
 class NoRequiredSchemaBuilder(SchemaBuilder):
-    EXTRA_STRATEGIES = (NoRequiredObj,)
+    EXTRA_STRATEGIES = (NoRequiredObj, IntegerToNumber)
 
 
 # This type is inferred from the genson lib, but there is no alias provided for it - creating it here for type safety
 InferredSchema = Dict[str, Union[str, Any, List, List[Dict[str, Union[Any, List]]]]]
 
+SchemaNode = Union[dict, dict[str, str, Any, list, list[dict[str, Any, list]]]]
```

Comment:
I'm a little confused by this type definition. what is a `dict[str, str, Any, list, list[dict[str, Any, list]]]` ?

Reply:
That's a leftover from some testing, the `InferredSchema` above is correct already - I removed it